internal/service/ec2/vpc_endpoint.go

```diff
@@ -272,6 +278,18 @@ func resourceVPCEndpointCreate(ctx context.Context, d *schema.ResourceData, meta
 		input.SubnetIds = flex.ExpandStringValueSet(v.(*schema.Set))
 	}
 
+	if v, ok := d.GetOk("service_region"); ok {
+		_, err := conn.DescribeVpcEndpointServices(ctx, &ec2.DescribeVpcEndpointServicesInput{
```

Comment:
Not sure you need to do the describe here, the service will error reasonably anyhow if the service isn't available in the region and we're already in apply here.

Reply:
I checked both (removing the describe function and leaving it) and it had the same effect so I removed it. Thank you for the feedback 👍